applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -104,7 +204,73 @@ handle(Data, Call) ->
 
     {'ok', C1} = cf_exe:get_call(Call),
     Updates = [{fun whapps_call:kvs_store/3, 'dynamic_cid', CID}
-               ,{fun whapps_call:set_caller_id_number/2, CID}
+	      ,{fun whapps_call:set_caller_id_number/2, CID}
               ],
     cf_exe:set_call(whapps_call:exec(Updates, C1)),
     cf_exe:continue(Call).
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec should_restrict_call(wh_json:object(), whapps_call:call(), ne_binary()) -> boolean().
+should_restrict_call(Data, Call, Number) ->
+    case wh_json:is_true(<<"enforce_call_restriction">>, Data, 'false') of
+        'false' -> 'false';
+        'true' -> should_restrict_call_by_account(Call, Number)
+    end.
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec should_restrict_call_by_account(whapps_call:call(), ne_binary()) -> boolean().
+should_restrict_call_by_account(Call, Number) ->
+    AccountId = whapps_call:account_id(Call),
+    AccountDb = whapps_call:account_db(Call),
+    case couch_mgr:open_cache_doc(AccountDb, AccountId) of
```

Comment:
Let me dig at this.  I copied this bit from "DISA" without really understanding the internals.
tx